classes/ThankYous/ThankYousRepository.php

```diff
@@ -0,0 +1,1121 @@
+			$this->db->query($query);
+		} else
+		{
+			$query = $this->query_factory->GetQueryInsert(self::THANKED_TABLE, $db_fields);
+			$this->db->query($query);
+			$id = $this->db->insertId();
+		}
+
+		return $id;
+	}
+
+	/**
+	 * Deletes all of a Thank You's Thankees.
+	 *
+	 * @param int $thank_you_id
+	 */
+	private function DeleteThankYouThankees(int $thank_you_id)
+	{
+		$query_string = "DELETE FROM " . self::THANKED_TABLE . " WHERE item_id=int:thank_you_id";
+
+		$this->db->query($query_string, $thank_you_id);
+	}
+
+	/**
+	 * Saves a Thank You's thanked User.
+	 * Due to the hard dependency on the Thank You,it is recommended that a check has been done for the the Thank You
+	 * with the given ID prior to calling this.
+	 *
+	 * @param int $thank_you_id
+	 * @param int $user_id
+	 */
+	private function SaveThankYouUser(int $thank_you_id, int $user_id)
+	{
+		$db_fields = [
+			'int:thanks_id' => $thank_you_id,
+			'int:user_id'   => $user_id
+		];
+
+		$query = $this->query_factory->GetQueryInsert(self::THANKED_USERS_TABLE, $db_fields);
+		$this->db->query($query);
+	}
+
+	/**
+	 * Delete all of a Thank You's Users.
+	 *
+	 * @param int $thank_you_id
+	 */
+	private function DeleteThankYouUsers(int $thank_you_id)
+	{
+		$query_string = "DELETE FROM " . self::THANKED_USERS_TABLE . " WHERE thanks_id=int:thank_you_id";
+		$this->db->query($query_string, $thank_you_id);
+	}
+
+	private function QueryJoinThankYouToTagged(QueryBuilder $query)
+	{
+		$query->AddJoin(self::THANK_YOU_TABLE, self::THANK_YOU_TAGS_TABLE, self::THANK_YOU_TAGS_TABLE, self::THANK_YOU_TABLE . ".id = " . self::THANK_YOU_TAGS_TABLE . ".item_id AND aggregation_id = " . self::AGGREGATION_ID);
+	}
+
+	private function QueryJoinThankedUsersToTagged(QueryBuilder $query)
+	{
+		$query->AddJoin(self::THANKED_USERS_TABLE, self::THANK_YOU_TAGS_TABLE, self::THANK_YOU_TAGS_TABLE, self::THANKED_USERS_TABLE . ".thanks_id = " . self::THANK_YOU_TAGS_TABLE . ".item_id AND aggregation_id = " . self::AGGREGATION_ID);
+	}
+
+	/**
+	 * @param QueryBuilder $query
+	 * @param int[]        $date_range
+	 */
+	private function QueryAddCreatedBetweenFilter(QueryBuilder $query, array $date_range)
+	{
+		$date_range = $this->utility->FormatDateRange($date_range);
+
+		$lower_date = $date_range[0] ?? null;
+		$upper_date = $date_range[1] ?? null;
+
+		if (!isset($lower_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Lower Date not found at offset 0");
+		}
+
+		if (!isset($upper_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Upper Date not found at offset 1");
+		}
+
+		if (!is_int($lower_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Lower Date is not an integer");
+		}
+
+		if (!is_int($upper_date))
+		{
+			throw new InvalidArgumentException("Failed to Add Created Between Filter to Query, Upper Date is not an integer");
+		}
```

Comment:
Could be consolidated into two throws by checking `isset` and `is_int` together

Reply:
Fixed in d18ed3f0bbd1477b87f3a3f4b5c2d902859dd250